driver/src/contracts/erc20.rs

```diff
@@ -0,0 +1,63 @@
+//! This module implements the ERC20 smart contract interface as well a trait
+//! and implementation to read it from the block chain.
+
+use super::Web3;
+use crate::models::TokenInfo;
+use crate::util::FutureWaitExt;
+use anyhow::Result;
+use ethcontract::H160;
+use lazy_static::lazy_static;
+use std::collections::HashMap;
+
+/// Trait for reading ERC20 token information from the block chain.
+#[cfg_attr(test, mockall::automock)]
+pub trait TokenReading {
+    /// Retrieve token information by address.
+    fn read_token_info(&self, token_address: H160) -> Result<TokenInfo>;
+}
+
+include!(concat!(env!("OUT_DIR"), "/erc20_detailed.rs"));
+
+/// Web3 `TokenReading` implementation.
+pub struct TokenReader {
+    web3: Web3,
+}
+
+impl TokenReader {
+    /// Create a new token reader from a web3 instnace.
+    #[allow(dead_code)]
+    pub fn new(web3: Web3) -> Self {
+        TokenReader { web3 }
+    }
+}
+
+lazy_static! {
+    static ref SYMBOL_OVERRIDES: HashMap<String, String> = hash_map! {
+        "WETH" => "ETH".to_owned(),
+    };
+}
+
+impl TokenReading for TokenReader {
+    /// Retrieve token information by address.
+    fn read_token_info(&self, token_address: H160) -> Result<TokenInfo> {
```

Comment:
Regarding Felix comment:
1) Performance issue:
If ee could do a promise all for many token addresses, and if do these requests only once a day (potentially even only at the instance start), then I would guess that performance is not an issue.

2) I would like to get all possible symbol information into the json-instance description. Even if some is fake, generally, I think that it can be helpful during debugging. 
But for sure, we should have a hardcoded list between SYMBOL and "real token", and only for tokens appearing in the list, I would propose to get a price feed. This would also allow us an easy identification of fake token symbols during the instance investigation.